metadata-checker/popup/metadata.js

```diff
@@ -0,0 +1,138 @@
+function onMetaData(metaData) {
+  metaData = metaData[0];
+
+  var shown = false;
+
+  for (prop in metaData) {
+    value = metaData[prop];
+    var section = null;
+    if (!prop.match(/^og:/) && !prop.match(/^twitter:/) && !prop.match(/^fb:/))
+      continue;
+    if (!shown) {
+      document.getElementById("metadata").setAttribute("class", "");
+      document.getElementById("nothing-to-see").setAttribute("class", "hidden");
+    }
+    let ul = document.getElementById("metadata-list");
+    let li = document.createElement("li");
+    let tt = document.createElement("tt");
+    tt.appendChild(document.createTextNode(prop));
+    li.appendChild(tt);
+    li.appendChild(document.createTextNode(": "+ value));
+    ul.appendChild(li);
+  }
+
+  // see http://ogp.me/
+  if (metaData["og:url"] || metaData["og:title"] || metaData["og:description"]) {
+    let container = document.getElementById("og-preview");
+    container.setAttribute("class", "");
+    //XXX: handle arrays ?
+    var image = metaData["og:image"] || metaData["og:image:src"];
+    // TODO: fixup "//..." urls
+    // cf. https://css-tricks.com/essential-meta-tags-social-media/
+    //if (image && image.match(/^\/\/:/))
+    //  image = (new URL(tabUrl)).protocol + image;
+    if (image) {
+      let div = document.createElement("div");
+      let img = document.createElement("img");
+      img.setAttribute("style", "max-width: 100px; max-height: 100px;");
```

Comment:
I'm not sure why you are setting this as an inline style, rather than in the stylesheet.